src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -880,20 +967,20 @@ private void changeScreenContent(String screenName, String sourceName) throws Ex
 	 * @return return true if mixed layer, false: single layer
 	 * @throws Exception Throw exception when fail to get JsonNode or split string
 	 */
-	private boolean checkLayerType(int currentScreenDestId) throws Exception {
+	private boolean checkLayerType(int currentScreenDestId, int layerId) throws Exception {
 		JsonNode screenDestContent = getScreenDestContent(currentScreenDestId);
 		JsonNode layerNode = screenDestContent.get(BarcoE2Constant.LAYERS);
-		String firstLayerName = layerNode.get(0).get(BarcoE2Constant.NAME).asText();
-		if (layerNode.get(0).get(BarcoE2Constant.NAME) == null) {
+		String firstLayerName = layerNode.get(layerId).get(BarcoE2Constant.NAME).asText();
```

Comment:
layerNode.get(layerId) and layerNode.get(layerId+1) can be extracted and reused

Reply:
Fixed